addons/sale/models/sale_order.py

```diff
@@ -611,6 +627,56 @@ def _compute_amount_undiscounted(self):
                 total += (line.price_subtotal * 100)/(100-line.discount) if line.discount != 100 else (line.price_unit * line.product_uom_qty)
             order.amount_undiscounted = total
 
+    @api.depends('client_order_ref', 'date_order', 'origin', 'partner_id', 'tax_totals')
+    def _compute_duplicated_order_ids(self):
+        order_to_duplicate_orders = self._fetch_duplicate_orders()
+        for order in self:
+            order.duplicated_order_ids = [Command.set(order_to_duplicate_orders.get(order.id, []))]
+
+    def _fetch_duplicate_orders(self):
+        """ Fectch duplicated orders.
+
+        :return: Dictionary mapping order to it's related duplicated orders.
+        :rtype: dict
+        """
+        orders = self.filtered('client_order_ref')
+        if not orders:
```

Comment:
```suggestion
        if not orders.filtered('id'):
```
Even if a ref is given, we shouldn't search duplicated orders for orders which are not saved.